client/tsconfig.json

```diff
@@ -0,0 +1,31 @@
+{
+  "extends": "../tsconfig.base.json" /* Path to base configuration file to inherit from. Requires TypeScript version 2.1 or later. */,
+  "include": [
+    "./**/*"
+  ] /* Specifies a list of glob patterns that match files to be included in compilation. If no 'files' or 'include' property is present in a tsconfig.json, the compiler defaults to including all files in the containing directory and subdirectories except those specified by 'exclude'. Requires TypeScript version 2.0 or later. */,
+  "files": [
+    "../package.json",
+    "../src/types/socket-events.d.ts",
+    "./js/helpers/fullnamemap.json",
+    "./js/helpers/simplemap.json",
+    "../src/helper.ts"
+  ] /* If no 'files' or 'include' property is present in a tsconfig.json, the compiler defaults to including all files in the containing directory and subdirectories except those specified by 'exclude'. When a 'files' property is specified, only those files and those specified by 'include' are included. */,
+  // "exclude": [],
+  "compilerOptions": {
+    "sourceMap": false /*Create source map files for emitted JavaScript files. See more: https://www.typescriptlang.org/tsconfig#sourceMap */,
+    "jsx": "preserve" /* Specify what JSX code is generated. */,
+    "lib": ["DOM", "DOM.Iterable", "ESNext"],
+
+    // https://v2.vuejs.org/v2/guide/typescript.html?redirect=true#Recommended-Configuration
+    // this aligns with Vue's browser support
+    "target": "es5",
```

Comment:
We don't care about Vue's older support, we have a higher target in browserlists, webpack and eslint.

There's quite a few places where ES target is set, which is a bit annoying.